games/devtest/mods/unittests/misc.lua

```diff
@@ -80,3 +80,20 @@ unittests.register("test_punch_node", function(_, pos)
 	minetest.remove_node(pos)
 	-- currently failing: assert(on_punch_called)
 end, {map=true})
+
+local function test_compress()
+	-- This text should be compressible, to make sure the results are... normal
+	local text = "The\000 icey canoe couldn't move very well on the\128 lake. The\000 ice was too stiff and the icey canoe's paddles simply wouldn't punch through."
+	local zstd_magic = string.char(0x28, 0xB5, 0x2F, 0xFD)
```

Comment:
Maybe a stupid question: Why not cut off the magic number to save the 4 bytes?

Reply:
Because a misdetection will break the DEFLATE side of the unit test, and will break any future compression algorithms that get added.

Reply:
Does this mean that if no algorithm is given, the code falls back to the magic number?